aya/src/obj/mod.rs

```diff
@@ -236,10 +254,30 @@ impl Object {
                 bpf_obj
                     .symbols_by_index
                     .insert(symbol.index().0, sym.clone());
+
+                if symbol.is_global() || symbol.kind() == SymbolKind::Data {
+                    bpf_obj.symbol_offset_by_name.insert(name, symbol.address());
+                }
             }
         }
 
+        // .BTF and .BTF.ext sections must be parsed first
+        // as they're required to prepare function and line information
+        // when parsing program sections
+        if let Some(s) = obj.section_by_name(".BTF") {
+            bpf_obj.parse_section(Section::try_from(&s)?)?;
+        }
+        if let Some(s) = obj.section_by_name(".BTF.ext") {
```

Comment:
should we parse .BTF.ext only if we parse .BTF first? If so, move inside if above. If not, the code that parses BTF.ext uwnrap()s if self.btf is None, so that needs to be fixed.